core/src/main/java/com/linecorp/armeria/common/ExceptionSampler.java

```diff
@@ -0,0 +1,37 @@
+package com.linecorp.armeria.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.function.Function;
+
+import org.jctools.maps.NonBlockingHashMap;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.util.Sampler;
+
+final class ExceptionSampler implements Sampler<Class<? extends Throwable>> {
+
+    private final Map<Class<? extends Throwable>, Sampler<Class<? extends Throwable>>> samplers =
+            new NonBlockingHashMap<>();
+    private final Function<? super Class<? extends Throwable>,
+            ? extends Sampler<Class<? extends Throwable>>> samplerFactory;
+    private final String spec;
+
+    ExceptionSampler(String spec) {
+        samplerFactory = unused -> Sampler.of(spec);
```

Comment:
Is there any chance that a user wants to set different specs for different exceptions?

Reply:
I'm not sure. Perhaps we can consider that when someone asks for it? :wink:

Reply:
Added a comment about this idea to `Flags`